client/components/data/query-canonical-theme/index.jsx

```diff
@@ -0,0 +1,37 @@
+/**
+ * External dependencies
+ */
+import React, { PropTypes } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import QueryTheme from 'components/data/query-theme';
+import { isWpcomTheme, isWporgTheme } from 'state/themes/selectors';
+
+const QueryCanonicalTheme = ( { siteId, themeId, isWpcom, isWporg } ) => (
+	<div>
+		<QueryTheme themeId={ themeId } siteId="wpcom" />
```

Comment:
Tricky. I guess checking `getCanonicalTheme` doesn't make that much sense either, since we can't know if the theme exists at all...